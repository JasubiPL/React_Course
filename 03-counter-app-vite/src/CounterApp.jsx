import { useState } from "react"

export const CounterApp = ({ value }) =>{

  const [counter, setCounter] = useState( value )


  return(
    <>
      <h1>Counter App</h1>
      <button onClick={() => setCounter( counter + 1 )}> +1 </button>
      <button onClick={() => setCounter( counter - 1 )}> -1 </button>
      <button onClick={() => setCounter( value )}> Reset </button>
      <p>{ counter }</p>
    </>
  )
}