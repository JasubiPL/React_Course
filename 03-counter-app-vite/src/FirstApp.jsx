import PropTypes from 'prop-types'

export const FirstApp = ( { title, subtitle, name } ) =>{

  return(
    <>
      <h1 data-testid='test-title'>{ title }</h1>
      <p>{ subtitle }</p>
      <p>{ name }</p>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

FirstApp.defaultProps = {
  name:'Jasubi',
  subtitle: 'No hay subtitulo'
  // title: 'No hay Titulo',
}