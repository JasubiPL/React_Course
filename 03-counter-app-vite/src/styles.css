html, body{
  background-color: #21232A;
  color: #fff;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button{
  font-size: 1.5rem;
  padding: 10px 20px;
  border-radius: 5px;
  background-color: #264ddb;
  border: none;
  color: #fff;
}

button:active{
  background-color: #2699db;
}