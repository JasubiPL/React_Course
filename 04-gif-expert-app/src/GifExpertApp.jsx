import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Javascript'])

  const onAddCategories = (submittedCategory) =>{
    if(categories.includes(submittedCategory)) return;
    
    setCategories([ submittedCategory, ...categories ])
  }

  const clean = () =>{
    setCategories([])
  }

  return(
    <>
      <h1>Gif Expert App</h1>

      <AddCategory onAddCategories={onAddCategories} categories={categories} cleanCategories={clean}/>

      { 
        categories.map(category => (
          <GifGrid key={category} category={category}/>
        )) 
      }
    </>
  )
}