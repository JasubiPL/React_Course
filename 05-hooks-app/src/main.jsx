import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
// import { HooksApp } from './HooksApp'
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { FormWithCustomeHook } from './02-useEffect/FormWithCustomeHook'
import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
import { FocusScreen } from './04-useRef/FocusScreen'
import { Layout } from './05-useLayoutEffect/Layout'

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
  // <MultipleCustomHooks />
  // <FocusScreen />
  <Layout />

    
  // </React.StrictMode>,
)
