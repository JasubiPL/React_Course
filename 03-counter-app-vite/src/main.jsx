import React from "react";
import ReactDOM from "react-dom/client";
// import { HelloWordApp } from "./HelloWordApp";
import { FirstApp } from "./FirstApp";
import './styles.css'
import { CounterApp } from "./CounterApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <FirstApp title="Titulo de la App" />
    {/* <CounterApp value={ 20 }/> */}
  </React.StrictMode>
)