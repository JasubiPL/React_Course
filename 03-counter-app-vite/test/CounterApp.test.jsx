import { render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Probando el componente CounterApp', () => {

  const value = 100
  test('Debe hacer match com el snapshot', () => {
  const { container } = render ( <CounterApp /> )

  expect( container ).toMatchSnapshot()
  });

  test('Debe mostrar un valor inicial de 100', () => {
    render ( <CounterApp value={value}/> )
    expect(parseInt(screen.getByText(value).innerHTML)).toBe(100)
  });
});