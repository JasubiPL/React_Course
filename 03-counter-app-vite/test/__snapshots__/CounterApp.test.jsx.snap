// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente CounterApp Debe hacer match com el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
  <p />
</div>
`;
